convert-macros/src/lib.rs

```diff
@@ -0,0 +1,100 @@
+use std::iter;
+
+use manyhow::{ensure, manyhow, Result};
+use proc_macro_crate::{crate_name, FoundCrate};
+use quote::{format_ident, quote, ToTokens};
+use syn::{parse_quote, Attribute, DeriveInput, Path};
+
+/// Tries to resolve the path to `extism_convert` dynamically, falling back to feature flags when unsuccessful.
+fn convert_path() -> Path {
+    match crate_name("extism") {
+        Ok(FoundCrate::Name(name)) => {
+            let ident = format_ident!("{name}");
+            parse_quote!(::#ident::convert)
+        }
+        Ok(FoundCrate::Itself) => parse_quote!(crate::core),
+        Err(_) => match crate_name("extism-convert").or_else(|_| crate_name("extism_pdk")) {
+            Ok(FoundCrate::Name(name)) => {
+                let ident = format_ident!("{name}");
+                parse_quote!(::#ident)
+            }
+            Ok(FoundCrate::Itself) => parse_quote!(crate),
+            Err(_) => match () {
+                () if cfg!(feature = "extism-path") => parse_quote!(::extism::convert),
+                () if cfg!(feature = "extism-pdk-path") => parse_quote!(::extism_pdk),
+                _ => parse_quote!(::extism_convert),
+            },
+        },
+    }
+}
```

Comment:
This feature based fallback logic is likely redundant. I implemented it this way in a similar multicrate project but don't remember the issues this solved: https://github.com/khonsulabs/bonsaidb/blob/8c9f47f97754e7613a18a24c092cb48423778acf/crates/bonsaidb-macros/src/lib.rs#L35-L61